tests/testsuite/update.rs

```diff
@@ -1659,3 +1659,56 @@ fn update_with_missing_feature() {
         )
         .run();
 }
+
+#[cargo_test]
+fn update_incompatible() {
+    Package::new("compatible", "1.0.0").publish();
+    Package::new("incompatible", "1.0.0").publish();
+    Package::new("pinned", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                compatible = "1.0"
+                incompatible = "1.0"
+                pinned = "<99.0"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("check").run();
+
+    Package::new("compatible", "1.0.1").publish();
+    Package::new("incompatible", "1.0.1").publish();
+    Package::new("pinned", "1.0.1").publish();
+
+    Package::new("compatible", "2.0.0").publish();
+    Package::new("incompatible", "2.0.0").publish();
+    Package::new("pinned", "2.0.0").publish();
+
+    p.cargo("update --incompatible")
+        .with_status(1)
+        .with_stderr_contains("[ERROR] unexpected argument '--incompatible' found")
```

Comment:
Let's start with the name `--breaking`

Incompatible is a broader term that can be "incomptible with this version req" and not just "semver incompatible".  "Breaking" is semantically more tied to semver

Reply:
(and it is easier to add a short flag for)

Reply:
Changed to `--breaking`. Although I will have to update this initial test as well when rewriting the Git history.

Reply:
Before you next want me to look at this, could you clean up the history?

Reply:
Done.